src/services/Search.php

```diff
@@ -396,7 +396,7 @@ private function _indexElementKeywords(int $elementId, string $attribute, string
 
         // Insert/update the row in searchindex
         $db->createCommand()
-            ->insert(Table::SEARCHINDEX, $columns, false)
+            ->upsert(Table::SEARCHINDEX, $columns, false)
```

Comment:
@brandonkelly, we may want to pass `true` to `$updateColumns` instead of `false`, thoughts?
```
The column data (name => value) to be updated if they already exist. If true is passed, the column data will be updated to match the insert column data. If false is passed, no update will be performed if the column data already exists.
```

The newer data triggered by the save is likely more desirable than the older data.

Reply:
Thinking about it more I like sticking with false, I really just want it to fail gracefully, as the user experience for jobs that continually fail is quite bad (it slows Craft's dashboard down for the end user).